### PR TITLE
fix "`mv` fails transfers between dirs"

### PR DESCRIPTION
**Pull request by rivy**
* includes test which demonstrates the problem 

Note that this is on top of the improved testing introduced in [PR#1281](https://github.com/uutils/coreutils/pull/1281).

### CODE REVIEW

File: src/mv/mv.rs [hunk 1]
```diff
@@ -360,6 +366,18 @@ fn rename(from: &PathBuf, to: &PathBuf, b: &Behaviour) -> Result<()> {
         }
     }
 
+    // "to" may no longer exist if it was backed up
+    if to.exists() && to.is_dir() {
+        // normalize behavior between *nix and windows
+        if from.is_dir() {
+            if is_empty_dir(to) {
+                try!(fs::remove_dir(to))
```
Comment (Arcterus) [inline src/mv/mv.rs]: Instead of adding another `try!` here, I'd prefer if you just change all the `try!`s to `?`.

File: src/mv/mv.rs [hunk 2]
```diff
@@ -405,3 +423,12 @@ fn existing_backup_path(path: &PathBuf, suffix: &str) -> PathBuf {
         simple_backup_path(path, suffix)
     }
 }
+
+fn is_empty_dir(path: &PathBuf) -> bool {
+    match fs::read_dir(path) {
+        Ok(contents) => {
+            return contents.peekable().peek().is_none();
+            },
```
Comment (Arcterus) [inline src/mv/mv.rs]: Extra indent

Comment (rivy) [reply src/mv/mv.rs]: I'll fix this, but, in addition, I just ran a `rustfmt` against the repo, and I was going to add a PR for it.

Maybe a `cargo fmt -- --check` could be added as a CI test.

File: src/mv/mv.rs (hunk 1 above)
Comment (rivy) [reply src/mv/mv.rs]: Ok, I'll look at refactoring that idiom.

**Comment by rivy** [comment]
I'll be happy to rebase all my PRs (they had to be based upon the initial PR in order to use the CI testing). It'll take me a just a little bit of time.

File: src/mv/mv.rs (hunk 2 above)
Comment (rivy) [reply src/mv/mv.rs]: Fixed.

**Comment by rivy** [comment]
@Arcterus , I rebased all of the basic PRs (#1304, #1305, #1315, #1316, #1317, #1319) and did some quick prelimary testing which looks fine (CI is still pending).

File: src/mv/mv.rs (hunk 1 above)
Comment (rivy) [reply src/mv/mv.rs]: Changed as requested.
Tests all pass locally, CI pending.

**Comment by rivy** [comment]
@Arcterus ,

Will you look at this and the other PRs mentioned (#1304, #1305, #1315, #1316, #1317, #1319)?
I think they are all relatively straight-forward fixes.
And, I don't want them to get stale and have to all be rebased again.
I'm happy to make changes, if needed.

**Comment by rivy** [comment]
@Arcterus , @cnd , I'm checking back in these PRs. Any issues that need to be fixed or are they mergeable?

**Comment by Arcterus** [comment]
Sorry it's taken so long, I'm gonna go through these.

**Pull request merged**